api/src/lib.rs

```diff
@@ -122,6 +122,10 @@ impl PostieApi {
 
         let res_str = res.text().await?;
         let res_json = serde_json::from_str(&res_str).unwrap_or_default();
+
+        let mut db = PostieDb::new().await;
```

Comment:
Not 100%, but since the Gui struct has an instance of a PostieApi as part of its Default implementation we could possibly pass that PositieApi into the `makeRequest` method?

So the function signature would be `pub async fn make_request(&self, input: HttpRequest) -> Result<Value, Box<dyn Error>> {`

and in theory PostieApi is available at self.api?